File: src/cn/yiiguxing/plugin/translate/action/ShowTranslationDialogAction.java
package cn.yiiguxing.plugin.translate.action;

import cn.yiiguxing.plugin.translate.TranslationUiManager;
import cn.yiiguxing.plugin.translate.ui.Icons;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.project.DumbAware;

public class ShowTranslationDialogAction extends AnAction implements DumbAware {

    public ShowTranslationDialogAction() {
        super(Icons.Translate);
        setEnabledInModalContext(true);
    }

    @Override
    public void actionPerformed(AnActionEvent e) {
        if (ApplicationManager.getApplication().isHeadlessEnvironment())
            return;

        TranslationUiManager.getInstance().showTranslationDialog(e.getProject());
    }

}

File: src/main/kotlin/cn/yiiguxing/plugin/translate/action/ShowTranslationDialogAction.kt
package cn.yiiguxing.plugin.translate.action

import cn.yiiguxing.plugin.translate.adaptedMessage
import cn.yiiguxing.plugin.translate.message
import cn.yiiguxing.plugin.translate.service.TranslationUIManager
import cn.yiiguxing.plugin.translate.util.SelectionMode
import cn.yiiguxing.plugin.translate.util.Settings
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.application.ApplicationManager

/**
 * 显示翻译对话框动作
 */
class ShowTranslationDialogAction : TranslateAction(true) {

    init {
        isEnabledInModalContext = true
        templatePresentation.text = adaptedMessage("action.ShowTranslationDialogAction.text")
        templatePresentation.description = message("action.ShowTranslationDialogAction.description")
    }

    override val selectionMode: SelectionMode
        get() = Settings.autoSelectionMode

    // should be always available
    override fun update(e: AnActionEvent) {}

    override fun actionPerformed(e: AnActionEvent) {
        if (ApplicationManager.getApplication().isHeadlessEnvironment) {
            return
        }

        TranslationUIManager.showDialog(e.project)
        if (Settings.takeWordWhenDialogOpens) {
            super.actionPerformed(e)
        }
    }

}